### DISCUSSION TITLE
Rewrite Codes According to the New Features of C#

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#123] [General]
Hi,

The title of book is **Professional** C# **2021**, Isn't it better to rewrite the codes according to the new features of C#? for example, Page 296 - CHAPTER 11 Tasks and asynchronous Programming, You wrote:

`private readonly static Dictionary<string, string> names = new Dictionary<string, string>();`

It can be 
`private readonly static Dictionary<string, string> names = new();`

However, the codes in [THIS](https://github.com/ProfessionalCSharp/ProfessionalCSharp2021/blob/main/1_CS/Tasks/TaskFoundations/Program.cs) file are in a new style but in the book, they are in the old style.

**Comment by christiannagel** [comment]
You are right that new features of C# should be used in this book - and they are. 
With nearly every code sample, **target-typed new** is used when the type is clear (from the left side). However, it's easy to miss this in a few places - like the one you mention in Chapter 11.

I'm making an update to the updates.md file (see bd0150149d308e2a8664f5773b418ac125cc9af0)

**Reply by ShervanN** [reply]
Thanks
